pkg/auth/agent_local_authenticator.go

```diff
@@ -86,6 +86,21 @@ func (a *AgentLocalAuthenticator) AuthAgentAuth(token string) (interface{}, erro
 	}
 	a.log.Infof("Authenticating infraEnv %s JWT", infraEnvID)
 
+	expiryDateTime, err := gencrypto.ParseExpirationFromToken(token)
```

Comment:
Instead of calling this method, just do what ever you need to do with the existing `claims` variable here.
That should make the logic a bit more clear and prevent you from having to match error strings.